services/meta/client.go

```diff
@@ -676,15 +676,15 @@ func (c *Client) PruneShardGroups() error {
 	data := c.cacheData.Clone()
 	for i, d := range data.Databases {
 		for j, rp := range d.RetentionPolicies {
+			remainingshards := data.Databases[i].RetentionPolicies[j].ShardGroups[:0]
```

Comment:
This code is rarely called, so optimizations are not really a concern here.  I understand that we can share the memory, but I would favor readability here over a premature optimization.  Thanks!